OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -222,6 +221,24 @@ const OpenSim::PhysicalFrame& Joint::getParentFrame() const
     return getConnector<PhysicalFrame>("parent_frame").getConnectee();
 }
 
+Coordinate::MotionType Joint::getMotionType(CoordinateIndex cix) const
+{
+    OPENSIM_THROW_IF(cix >= _motionTypes.size(), Exception,
+        "Joint::getMotionType() given an invalid CoordinateIndex");
+    return _motionTypes[cix];
+}
+
+void Joint::setMotionType(CoordinateIndex cix, Coordinate::MotionType mt)
+{
+    OPENSIM_THROW_IF(cix >= numCoordinates(), Exception,
+        "Joint::setMotionType() for an invalid CoordinateIndex");
+    if (_motionTypes.size() <= cix)
```

Comment:
`numCoordinates() == (cix+1)` so I can replace the to condition to be `_motionTypes.size() < numCoordinates()`.  I agree that would be easier to understand. The important thing is that the array must be resized to accommodate the `MotionType` of Coordinates the `Joint` is adding to its list (`CoordinateSet`). I'll make the change and add comments. Thanks.